### ISSUE TITLE
Report error for mismatch of record parameter and explicit property

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#44694]
Should report an error for mismatch in `X`:
```C#
data class C(object X, int Y) // error: cannot assign to C.X
{
    public int X { get; }
    public object Y { get; }
}
```

**Comment by cston** [comment]
The following should also be reported as mismatches between parameters and explicit properties.
```C#
data class C(object P1, object P2, object P3)
{
    public object P1 { set { } }
    public static object P2 { get; set; }
    public ref object P3 => throw null;
}
```

**Comment by cston** [comment]
I believe the current behavior is by design. The synthesized constructor should not assign parameters to explicitly declared properties.
